pkg_security.Rmd

```diff
@@ -1,19 +1,73 @@
 # Package Development Security Best Practices
 
 ```{block, type='summaryblock'}
-This chapter isn't written yet. It should mention the elements outlined in [this bookdown](https://ropenscilabs.github.io/r-security-practices/) ([repo](https://github.com/ropenscilabs/r-security-practices)) started at the unconf 2017, or have a link to this book/a similar book when written.
+This chapter isn't complete yet. It includes [guidance about managing secrets in packages](#pkgsecrets) and [links for further reading](#furthersecreading).
 ```
 
-In the meantime, here are useful security resources:
+## Misc
 
-* [rOpenSci community call "Security for R"](https://ropensci.org/commcalls/2019-05-07/) (recording, slides, etc. see in particular [the list of resources](https://ropensci.org/blog/2019/04/09/commcall-may2019/#resources));
+We recommend you [secure your GitHub account with 2FA](https://help.github.com/articles/securing-your-account-with-two-factor-authentication-2fa/).
 
-* [the security-related projects of unconf18](https://ropensci.org/blog/2018/06/06/unconf18_recap_2/);
+If the web service your package wraps has either https or http, opt for https.
```

Comment:
AFAIK (!!) this is especially true if the query string contains the API key. When using https the query parameters will be encrypted (https://stackoverflow.com/q/323200), with http they are not. So an attacker could easily extract your key by sniffing your http traffic. 

@sckott [mentioned](https://github.com/ropensci/opencage/pull/99#issuecomment-574757014) that it is better to send the key as a request header rather than a query param, but I don't know the exact reason for that.

Reply:
@sckott why is it best practice to pass secrets via headers rather than the body?

Reply:
@sckott why is it best practice to pass secrets via headers rather than the body?

Reply:
it can go either way IF https is used, but a few reasons:
- a user might ask a question and share the url, including the key. it's harder to accidentally share a key when it's in the header i think at least. a user might also want to just copy/paste url into a browser, where that link about browser stuff does then apply. seems good to not help people shoot themselves in the foot, so to speak
- urls can only be so long. in the unusual case that there are some very long query parameters, and/or the key is very long you could run into URI too long errors